code/misc.h

```diff
@@ -235,6 +235,12 @@ typedef const struct SrcIdStruct {
 #define BS_BITFIELD(ty, s)      BITFIELD(ty ## Set, (s), ty ## LIMIT)
 
 
+/* Save and restore errno. See <design/thread-manager#.sol.thread.errno> */
+
+#define ERRNO_SAVE BEGIN int _mps_saved_errno = (errno); BEGIN
+#define ERRNO_RESTORE END; (errno) = _mps_saved_errno; END
+
```

Comment:
I had fun researching the reasons for the BEGIN/END constructs.
I'm a bit puzzled by parantheses around errno.  Is that defensive against whatever macro errno might be in some implementations?

Reply:
You're right, the parentheses are unnecessary: this macro can only be used after `#include <errno.h>` so `errno` should be defined correctly by the header. I'll add a note and remove the parentheses.